core/modules/css/DbInterfaceMySql.cc

```diff
@@ -0,0 +1,126 @@
+// -*- LSST-C++ -*-
+
+/*
+ * LSST Data Management System
+ * Copyright 2019 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "css/DbInterfaceMySql.h"
+
+// System headers
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <stdexcept>
+
+// Third-party headers
+#include "mysql/mysqld_error.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "css/CssError.h"
+#include "css/DbInterfaceMySql.h"
+#include "sql/SqlResults.h"
+#include "sql/SqlTransactionStd.h"
+
+using namespace std;
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.css.DbInterfaceMySql");
+
+} // namespace
+
+namespace lsst {
+namespace qserv {
+namespace css {
+
+
+class KvTransaction : public sql::SqlTransactionStd
+{
+public:
+    // Constructors
+    KvTransaction(sql::SqlConnection& conn)
+        : sql::SqlTransactionStd(conn) {}
+
+    // Destructor
+    ~KvTransaction() override = default;
+
+    /// Override to throw an appropriate exception.
+    void throwException(util::Issue::Context const& ctx, std::string const& msg) override {
+        throw CssError(ctx, msg + " mysql(" +std::to_string(errObj.errNo()) + " " + errObj.errMsg() + ")");
+    }
+};
+
+
+DbInterfaceMySql::DbInterfaceMySql(mysql::MySqlConfig const& mysqlConf)
+: _conn(mysqlConf) {
+}
+
+
+std::set<int> DbInterfaceMySql::getEmptyChunks(std::string const& dbName) {
+    std::string funcName("getEmptyChunksFromDb");
+    std::lock_guard<std::mutex> sync(_dbMutex);
+
+    KvTransaction trans(_conn);
+
+    /// This must match the name of the table generated by
+    /// dataloader.py DataLoader _makeEmptyChunksTable()
+    std::string emptyChunkTbl = dbName + "_EmptyChunks";
+
+    // run query
+    sql::SqlErrorObject errObj;
+    sql::SqlResults results;
+    std::string const query = "SELECT * FROM " + emptyChunkTbl;
```

Comment:
I understand that the current implementation of the method is built on assumption that the empty chunk list tables have a single column (`chunkId`, etc.) and that is the very first column in the schema. Although it may seem to be sufficient to `SELECT * ...` now this makes code more susceptible to the schema migrations. Hence, to protect this code from the migrations one may consider a small reinforcement like `SELECT chunkId ...`.